rp2040-hal/src/multicore.rs

```diff
@@ -165,10 +176,12 @@ impl<'p> Core<'p> {
                 stack_ptr.write(v);
             };
 
-            push(wrapper as usize);
+            // We don't want to drop this, since it's getting moved to the other core.
+            let mut entry = ManuallyDrop::new(entry);
+
+            // Push the arguments to `core1_startup` onto the stack.
             push(stack.as_mut_ptr() as usize);
-            push(entry[1]);
-            push(entry[0]);
+            push(&mut entry as *mut _ as usize);
```

Comment:
Isn't this a pointer to the stack of core0, which gets invalid as soon as `spawn` returns? But may still be accessed from `core1_startup`, later?
Even if `entry` is `ManuallyDrop` and so won't get dropped at the end of the method, its memory will still be invalidated.

Reply:
Yes, but core 0 waits for core 1 to signal that it's read from the pointer before returning: https://github.com/rp-rs/rp-hal/blob/0e5fdcfd9cbdd0efbf4af1315366a5b4c2e17b10/rp2040-hal/src/multicore.rs#L235-L236

Reply:
How would core1_startup access this after `spawn` returns?
We spin waiting for a signal from the other core before returning from `spawn`.
Entry is the entrypoint for core1, it can't send a reply until it has read it and started executing, and it would never be read again.

Reply:
Sorry, my reply was to jannic - apologies if it is confusing after Liam's comment

Reply:
You're right: I missed the fact that `let entry = unsafe { ManuallyDrop::take(entry) };` is the point where the pointer gets dereferenced, and the contents get copied to a new location on core1's stack.
(ie. the argument to `ManuallyDrop::take` is a reference, but the new `let entry` is no longer a reference but an owned type.)